inference-engine/src/transformations/include/transformations/convert_reduce_to_pooling.hpp

```diff
@@ -231,21 +231,34 @@ ngraph::matcher_pass_callback ConvertReduceBase::convert_reduce_to_pooling() {
             input.get_node_shared_ptr()->set_friendly_name(reduce->get_friendly_name() + "/pool");
             new_ops.push_back(input.get_node_shared_ptr());
         } else if (std::is_same<T, ngraph::opset1::ReduceSum>()) {
+            // Fallback to real type because of potential data loss in case of integer AVG Pool
+            bool fallback_to_real = input.get_element_type().is_integral();
+
+            if (fallback_to_real) {
+                input = std::make_shared<ngraph::opset1::Convert>(input, ngraph::element::f32);
```

Comment:
Just a note: this pass is applied after ConvertPrecision so plugin may get undesired f32 precision.

Reply:
Reasonable.  Do you know any real cases where it will lead to undesired effects? Does some one use this convert pass for FP32 data type?

I guess FP32 is a base precision for execution and supported by all plugins and this limitation will no affect product significantly.